src/python/pants_release/generate_github_workflows.py

```diff
@@ -19,6 +19,36 @@
 
 from pants.util.strutil import softwrap
 
+
+def action(name: str, node16_compat: bool = False) -> str:
+    # Versions of actions compatible with node16 and the `ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION` setting.
+    # glibc 2.17 is required to build manylinux_2014 wheels, but node.js does do not ship glibc 2.17 compatible
+    # binaries for node >= v17.
+    if node16_compat:
+        return {
+            "checkout": "actions/checkout@v3",
+            "upload-artifact": "actions/upload-artifact@v3",
+            "setup-go": "actions/setup-go@v4",
+        }[name]
```

Comment:
It would be nice to raise an error explaining that there is no node16 compatible version configured for the named action, instead of just a KeyError.

Reply:
Basically, a naive caller of this new method (likely via copypasta) will not immediately understand its failures, so it would be good to help.